src/libraries/System.Private.CoreLib/src/System/Threading/LazyInitializer.cs

```diff
@@ -272,6 +287,7 @@ private static T EnsureInitializedCore<T>([AllowNull] ref T target, ref bool ini
                 }
             }
 
+            NullabilityHelpers.SuppressNonNullAssignmentWarning(ref target);
```

Comment:
We should delete the TODO-NULLABLE comment below.  I'm also a little surprised the `!` on `target` below isn't suppressing the warning on target for [NotNull].